notebooks/concepts/Specviz from splot concept notebook.ipynb

```diff
@@ -0,0 +1,3636 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reg = specviz.get_selected_region()  #<- uses the \"most recent\" selection\n",
+    "reg2 = specviz.get_selected_region('Subset 2')  #<- uses glue's name for the subset\n",
+    "\n",
+    "all_regs = specviz.get_selected_region(all=True)  #<- this would be *all* the subsets combined together into a single SpectralRegion object, which is [ALREADY IMPLEMENTED] by SpectralRegion (but not specviz)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    Note that it is critical for some use cases that the above should yield spectral regions that are correct *even to the sub-pixel*.  That is, if the user does a region selection that ends between two pixels, that should be reflected in the SpectralRegion - this means using subset *states* in glue instead of the subset itself.\n",
+    "\n",
+    "    These selections can now be used with standard specutils analysis functions in a notebook setting (these are [ALREADY IMPLEMENTED]):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "specutils.analysis.line_flux(specviz.current_spectrum, specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.equivalent_width(specviz.current_spectrum, regions=specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.gaussian_fwhm(specviz.current_spectrum, specviz.get_selected_region())\n",
+    "\n",
+    "specutils.analysis.snr(specviz.current_spectrum, specviz.get_selected_region())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    In addition to region selection, there is value in having a way to extract a specific point in the displayed spectrum into the notebook. (Some examples are shown below) - while there are many potential solutions to this, the simplest is a stateful \"get location\" method.  E.g.,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spectral_coordinate, flux_coordinate = specviz.get_spectrum_location()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    This method would block, cause the cursor in the viz tool to change to some indicator cursor, and then un-block as soon as the user clicks somewhere in the viewer window, reporting out the spot the click."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. In the Desktop interface, relevant analysis functions like the above should be exposed as information plugins - when run this plugin would simply show the result of these functions as run in the latest `get_spectral_region` subset exactly as shown above but auto-update when the user creates a new/updates the current subset.\n",
+    "\n",
```

Comment:
I want to echo the desktop user interface requirements for the MAST interface version as well.  All of the splot functionalities you mention need to be available in the UI in addition to the notebook.  Our users won't have access to underlying notebooks.  We don't necessarily need all the functionality, maybe a subset or a simplified version. 
 We have a list of features that users wanted in their quicklook web interface.  See https://innerspace.stsci.edu/display/ASB/Rank+list+of+features, Functional Features - Visualization, Spectral Manipulation, and Spectral Analysis.

Reply:
Good point!  Will add some words on that.

Reply:
I'm also going to add a ticket to look through that list and compare here, but I'll do that in the ST Jira since it's not visible to those who might come across this issue from the outside.